### PR TITLE
add support for older CPU via a new container image

### LINKED ISSUE #18365: Docker image no longer works on older CPU (CPU does not support x86-64-v2)
After upgrading my Docker based minio installation from RELEASE.2023-10-25T06-33-25Z to RELEASE.2023-11-01T01-57-10Z the container refuses to start on [older CPU](https://ark.intel.com/content/www/us/en/ark/products/33927/intel-xeon-processor-e5420-12m-cache-2-50-ghz-1333-mhz-fsb.html).

## Expected Behavior

The minio container starts.

## Current Behavior

The minio container fails to start with the exit code 127 and the following error message: `Fatal glibc error: CPU does not support x86-64-v2`. No other output is provided.

## Steps to Reproduce (for bugs)

1. `docker run -it minio/minio:RELEASE.2023-11-01T01-57-10Z`

## Context

While I understand this is an old CPU, my understanding is that minio itself does not require these CPU instructions. Furthermore, the [binary](https://dl.min.io/server/minio/release/linux-amd64/minio.RELEASE.2023-11-01T01-57-10Z) used in the docker image works on the same machine so it seems the issue comes with the new base image. In fact, when trying to run the new base image I encounter the same error.

```sh
$ docker run registry.access.redhat.com/ubi9/ubi-minimal:9.2
Fatal glibc error: CPU does not support x86-64-v2
```

## Regression

Yes:
- [RELEASE.2023-11-01T01-57-10Z](https://github.com/minio/minio/releases/tag/RELEASE.2023-11-01T01-57-10Z)
- #18313
- #18329

## Your Environment

* Version used (`minio --version`):

```
minio version RELEASE.2023-11-01T01-57-10Z (commit-id=55e713db0a367f6cccee00af49f00c269e6ca619)
Runtime: go1.21.3 linux/amd64
License: GNU AGPLv3 <https://www.gnu.org/licenses/agpl-3.0.html>
Copyright: 2015-2023 MinIO, Inc.
```

* Server setup and configuration: N/A

* Operating System and version (`uname -a`):

`Linux mir 6.1.53-gentoo-x86_64-jmk-selinux #1 SMP PREEMPT_DYNAMIC Thu Sep 14 13:12:26 CEST 2023 x86_64 Intel(R) Xeon(R) CPU E5420 @ 2.50GHz GenuineIntel GNU/Linux`

* Hardware:

```
processor       : 0
vendor_id       : GenuineIntel
cpu family      : 6
model           : 23
model name      : Intel(R) Xeon(R) CPU           E5420  @ 2.50GHz
stepping        : 6
microcode       : 0x60b
cpu MHz         : 2500.082
cache size      : 6144 KB
physical id     : 0
siblings        : 8
core id         : 0
cpu cores       : 8
apicid          : 0
initial apicid  : 0
fpu             : yes
fpu_exception   : yes
cpuid level     : 10
wp              : yes
flags           : fpu de tsc msr pae mce cx8 apic sep mca cmov pat clflush acpi mmx fxsr sse sse2 ht syscall nx lm constant_tsc rep_good nopl cpuid tsc_known_freq pni est ssse3 cx16 sse4_1 hypervisor lahf_lm
bugs            : cpu_meltdown spectre_v1 spectre_v2 spec_store_bypass l1tf mds swapgs itlb_multihit mmio_unknown
bogomips        : 5000.02
clflush size    : 64
cache_alignment : 64
address sizes   : 38 bits physical, 48 bits virtual
power management:
```

### PR DESCRIPTION
**Pull request by harshavardhana**
## Community Contribution License
All community contributions in this pull request are licensed to the project maintainers
under the terms of the [Apache 2 license] (https://www.apache.org/licenses/LICENSE-2.0). 
By creating this pull request I represent that I have the right to license the 
contributions to the project maintainers under the Apache 2 license.

## Description
add support for older CPU via a new container image

## Motivation and Context
fixes #18365

## How to test this PR?
As per #18365 do not have the CPU in-house.

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

**Review by kannappanr — APPROVED** [review]
LGTM

File: Dockerfile.release.old_cpu [hunk 1]
```diff
@@ -0,0 +1,61 @@
+FROM golang:1.21-alpine as build
```
Comment (vadmeste) [inline Dockerfile.release.old_cpu:1]: This is the line that will allow to run this image in old CPUs ?

**Review by vadmeste — APPROVED** [review]
LGTM

File: Dockerfile.release.old_cpu (hunk 1 above)
Comment (harshavardhana) [reply Dockerfile.release.old_cpu:1]: No @vadmeste `ubi8` - ubi9 doesn't run on CPUs from 15yrs ago :-)

**Comment by rcmonitor** [comment]
How do I use this older image tag with helm-driven installation of MinIO?
I've got minio-operator and minio-tenant charts v7.1.1 installed and operator-sidecar containers fails with `Fatal glibc error: CPU does not support x86-64-v2`.
If I got it right, I should substitute image using `operator.sidecarImage.tag` value in values.yaml for minio-operator/minio-operator chart.
But, what tag to choose?
Here's a list of available tags for operator-sidecar image:
```$ regctl tag ls quay.io/minio/operator-sidecar
v6.0.0
v6.0.0-amd64
v6.0.0-arm64
v6.0.0-ppc64le
v6.0.0-s390x
v6.0.2
v6.0.2-amd64
v6.0.2-arm64
v6.0.2-ppc64le
v6.0.2-s390x
v7.0.0
v7.0.0-amd64
v7.0.0-arm64
v7.0.0-ppc64le
v7.0.0-s390x
v7.0.1
v7.0.1-amd64
v7.0.1-arm64
v7.0.1-ppc64le
v7.0.1-s390x
v7.1.0
v7.1.0-amd64
v7.1.0-arm64
v7.1.0-ppc64le
v7.1.0-s390x
```
Not a single one have a `-cpuv1` suffix...

**Pull request merged**